### PR TITLE
t-ronmoneta/sampleapp

### PR DESCRIPTION
**Pull request by ronmonetaMicro**
#### Pull Request Description

Golang SDK quickstart sample app 

---

#### Future Release Comment
Added Golang SDK quickstart sample app, in compliance with other SDK's sample apps

**Breaking Changes:**
- None

**Features:**
- None

**Fixes:**
- None

### CODE REVIEW

File: kusto/quickstart/utils/queries/queries.go [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+		iter, err = kustoClient.Query(ctx, databaseName, command)
+	}
+
+	if err != nil {
+		panic(fmt.Sprintf("Command execution failed: '%s'\n", err.Error()))
+	}
+	defer iter.Stop()
+
+	// .Do() will call the function for every row in the table.
+	err = iter.Do(
```
Comment (AsafMah) [inline kusto/quickstart/utils/queries/queries.go:30]: Use DoOnRowOrError

**Review by AsafMah — CHANGES_REQUESTED** [review]

File: kusto/ingest/file_options.go [hunk 2]
```diff
@@ -206,6 +206,11 @@
 	SingleJSON DataFormat = properties.SingleJSON
 )
 
+// GetFormat Translates a given string to it's DataFormat object
+func GetFormat(fName string) DataFormat {
```
[the remark this answers is not in the document]
Comment (AsafMah) [reply kusto/ingest/file_options.go:210]: I'll be more clear - remove it. Use the underlying method.

File: quickstart/sample_app.go [hunk 3]
```diff
@@ -0,0 +1,273 @@
...
+	// TODO: Change if needed. Disabled to prevent an existing batching policy from being unintentionally changed
+	if false && config.BatchingPolicy != "" {
+		waitForUserToProceed(fmt.Sprintf("Alter the batching policy for table '%s.%s'", config.DatabaseName, config.TableName))
+		alterBatchingPolicy(kustoClient, config.DatabaseName, config.TableName, config.BatchingPolicy)
+	}
+}
+
+// alterMergeExistingTableToProvidedSchema Alter-merges the given existing table to provided schema
+func alterMergeExistingTableToProvidedSchema(kustoClient *kusto.Client, databaseName string, tableName string, tableSchema string) {
+	command := kql.New(".alter-merge table ").AddTable(tableName).AddLiteral(" ").AddUnsafe(tableSchema)
```
Comment (AsafMah) [inline quickstart/sample_app.go:116]: Add comment about AddUnsafe

File: quickstart/sample_app.go [hunk 4]
```diff
@@ -0,0 +1,273 @@
...
+	queries.ExecuteCommand(kustoClient, databaseName, command)
+}
+
+// queryExistingNumberOfRows Queries the data on the existing number of rows
+func queryExistingNumberOfRows(kustoClient *kusto.Client, databaseName string, tableName string) {
+	command := kql.New("table(_table_name) | count")
+	params := kql.NewParameters().AddString("_table_name", tableName)
+	queryOptions := []kusto.QueryOption{kusto.QueryParameters(params)}
+
+	queries.ExecuteCommand(kustoClient, databaseName, command, queryOptions...)
```
Comment (AsafMah) [inline quickstart/sample_app.go:126]: Shorten to:
```go
	queries.ExecuteCommand(kustoClient, databaseName, command, kusto.QueryParameters(params))
```

File: quickstart/utils/ingestion/ingestion.go [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+		fmt.Print(i)
+		for j := 0; j <= i; j++ {
+			fmt.Print(".")
+		}
+		time.Sleep(time.Second)
+	}
+}
+
+// CreateIngestionOptions creates ingestion options: ingestion mapping and anything else necessary
+func CreateIngestionOptions(mappingName string, dataFormat ingest.DataFormat) []ingest.FileOption {
```
Comment (AsafMah) [inline quickstart/utils/ingestion/ingestion.go:30]: This is only used once and doesn't really save any lines, do you really need it?

File: quickstart/utils/queries/queries.go [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+	} else {
+		iter, err = kustoClient.Query(ctx, databaseName, command, options...)
+	}
+
+	if err != nil {
+		panic(fmt.Sprintf("Command execution failed: '%s'\n", err.Error()))
+	}
+	defer iter.Stop()
+
+	// .Do() will call the function for every row in the table.
```
Comment (AsafMah) [inline quickstart/utils/queries/queries.go:29]: Fix the comment

File: quickstart/utils/queries/queries.go [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+	err = iter.DoOnRowOrError(
+		func(row *table.Row, _ *kustoErrors.Error) error {
```
Comment (AsafMah) [inline quickstart/utils/queries/queries.go:31]: Don't ignore the error

**Review by AsafMah — COMMENTED** [review]
Some comments

File: kusto/ingest/file_options.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (ronmonetaMicro) [reply kusto/ingest/file_options.go:210]: I cant, properties is an internal package of ingest - so i cant import/call it from the sample app

File: quickstart/utils/ingestion/ingestion.go (hunk 5 above)
Comment (ronmonetaMicro) [reply quickstart/utils/ingestion/ingestion.go:30]: Not necessarily, was mostly done to maintain a similar structure to other SDKs, and to allow reusability.  should i remove it then?

File: kusto/ingest/file_options.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (AsafMah) [reply kusto/ingest/file_options.go:210]: Call it InferFormatFromFileName or something

**Review by AsafMah — COMMENTED** [review]
d

File: quickstart/utils/ingestion/ingestion.go (hunk 5 above)
Comment (AsafMah) [reply quickstart/utils/ingestion/ingestion.go:30]: I think we can remove it for now

**Review by AsafMah — APPROVED** [review]

**Pull request merged**